compiler/src/org.graalvm.compiler.lir.amd64/src/org/graalvm/compiler/lir/amd64/AMD64ControlFlow.java

```diff
@@ -283,9 +283,7 @@ public void emitCode(CompilationResultBuilder crb, AMD64MacroAssembler masm) {
             masm.jmp(scratchReg);
 
             // Inserting padding so that jump table address is 4-byte aligned
-            if ((masm.position() & 0x3) != 0) {
-                masm.nop(4 - (masm.position() & 0x3));
-            }
+            masm.align(4);
```

Comment:
I think you might want to align to 8 bytes if defaultTarget != null since each entry is 8 bytes wide.

Reply:
That's the alignment for the regular table switch, it hasn't changed. Here's the alignment for the hash table switch: https://github.com/oracle/graal/pull/895/files/7341ab80f998290de7310f393b4191a9f509813c#diff-a8baf9fe5005ee604d024bb2e35c173aR371